src/MKLSparse.jl

```diff
@@ -1,7 +1,9 @@
 module MKLSparse
 
+using MKL_jll
+
 function __init__()
-    ccall((:MKL_Set_Interface_Layer, :libmkl_rt), Cint, (Cint,), Base.USE_BLAS64 ? 1 : 0)
+    ccall((:MKL_Set_Interface_Layer, libmkl_rt), Cint, (Cint,), Base.USE_BLAS64 ? 1 : 0)
```

Comment:
I am a bit worried about this (for the future, not directly related to this PR). There has been ideas to use the 32-bit version for MKL with BLAS (https://github.com/JuliaComputing/MKL.jl/pull/35) but then all of a sudden, all your sparse matrix multiplications that use 64-bit integers would no longer hit the dispatch anymore...

Reply:
One issue is that the interface layer can be changed at runtime but this package defines its overload for a compilation constant (`BlasInt`). This is because this package does type piracy (https://github.com/JuliaSparse/MKLSparse.jl/issues/24). It would be easier if there was a separate `MKLSparse.matmul` and then we can just query what interface is currently used and convert the matrix if needed.

Reply:
I actually think that would be far safer! I'd say we get this PR in shape and then address that change later.